### PR TITLE
Fix front-end Helpers.ifHasPermissions

### LINKED ISSUE #1344: Frontend Helpers.ifHasPermissions broken

### PR DESCRIPTION
**Pull request by taylortom**
Fixes #1344

### CODE REVIEW

File: frontend/src/core/app/helpers.js [hunk 1]
```diff
@@ -87,48 +94,56 @@ define(function(require){
...
           } else {
             return block.inverse(this);
           }
         },
+
         selected: function(option, value){
             if (option === value) {
                 return ' selected';
             } else {
                 return ''
```
Comment (tomgreenfield) [inline frontend/src/core/app/helpers.js]: Missing semi-colon.

File: frontend/src/core/app/helpers.js [hunk 2]
```diff
@@ -221,8 +237,7 @@ define(function(require){
         },
```
Comment (tomgreenfield) [inline frontend/src/core/app/helpers.js]: Line 232, update to strict equality:
```js
url.indexOf('course/assets') === 0
```

File: frontend/src/core/app/helpers.js [hunk 3]
```diff
@@ -253,47 +268,45 @@ define(function(require){
...
```
Comment (tomgreenfield) [inline frontend/src/core/app/helpers.js]: Line 262 missing semi-colon.

File: frontend/src/core/app/helpers.js [hunk 4]
```diff
@@ -87,48 +94,56 @@ define(function(require){
...
             for (var i = 1; i <= n; ++i)
                 sum += block.fn(i);
             return sum;
         },
+
         t: function(str, options) {
             for (var placeholder in options.hash) {
               options[placeholder] = options.hash[placeholder];
             }
             return (window.polyglot != undefined ? window.polyglot.t(str, options) : str);
```
Comment (tomgreenfield) [inline frontend/src/core/app/helpers.js]: Update to strict equality:
```js
window.polyglot !== undefined
```

File: frontend/src/core/app/helpers.js [hunk 5]
```diff
@@ -87,48 +94,56 @@ define(function(require){
...
         },
+
         stripHtml: function(html) {
             return new Handlebars.SafeString(html);
         },
+
         bytesToSize: function(bytes) {
             if (bytes == 0) return '0 B';
```
Comment (tomgreenfield) [inline frontend/src/core/app/helpers.js]: Update to strict equality:
```js
bytes === 0
```

**Comment by canstudios-davidl** [comment]
Hi, any update on what's happening with this PR? I've just encountered this bug but see there's already a fix for it... PR looks a bit stale though.
Thanks

**Comment by taylortom** [comment]
@canstudios-davidl I'll pick this up for the next release

**Comment by canstudios-davidl** [comment]
Thanks @taylortom happy to help if needed

**Comment by canstudios-louisem** [comment]
Happy to approve once the conflicts are resolved.

File: frontend/src/core/helpers.js [hunk 6]
```diff
@@ -1,413 +1,336 @@
...
+        var selected = selectedValue == options[i] ? ' selected' : '';
+        html += '<option value="' + options[i] + '"' + selected + '>' + options[i] + '</option>';
+      }
+      return new Handlebars.SafeString(html);
+    },
+
+    pickCSV: function (list, key, separator) {
+      var vals = [];
+      separator = (separator && separator.length) ? separator : ',';
+      if (list && list.length) {
```
Comment (tomgreenfield) [inline frontend/src/core/helpers.js]: Looks like the logic hasn't been reversed here when returning early.

Anyhow, suggest refactoring to return right at the start of the function. Something like:
```js
pickCSV: function(list, key, separator) {
  if (!list || !list.length) {
    return '';
  }
  var vals = [];
  if (!separator || !separator.length) {
    separator = ',';
  }
  for (var i = 0, l = list.length; i < l; i++) {
    var val = list[i];
    var nestedVal = key && val[key];
    vals.push(nestedVal || val);
  }
  return vals.join(separator);
}
```

File: frontend/src/core/helpers.js [hunk 7]
```diff
@@ -1,413 +1,336 @@
...
+      var hasPermission = Origin.permissions.hasPermissions(permissions.split(','));
+      return hasPermission ? block.fn(this) : block.inverse(this);
+    },
+
+    ifMailEnabled: function(block) {
+      return Origin.constants.useSmtp === true ? block.fn(this) : block.inverse(this);
+    },
+
+    ifImageIsCourseAsset: function(url, block) {
+      var isCourseAsset = url.length !== 0 && url.indexOf('course/assets') == 0;
```
Comment (tomgreenfield) [inline frontend/src/core/helpers.js]: Please use triple equals: `url.indexOf('course/assets') === 0`

File: frontend/src/core/helpers.js [hunk 8]
```diff
@@ -1,413 +1,336 @@
...
 
-        ifMailEnabled: function(block) {
-          if (Origin.constants.useSmtp === true) {
-            return block.fn(this);
-          } else {
-            return block.inverse(this);
-          }
-        },
+    bytesToSize: function(bytes) {
+      if (bytes == 0) return '0 B';
```
Comment (tomgreenfield) [inline frontend/src/core/helpers.js]: Please use triple equals: `bytes === 0`.

**Pull request merged**